src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogWebhookBuildLogic.java

```diff
@@ -72,10 +60,30 @@ public JSONObject toJson(final BuildData buildData, final Run<?,?> run) {
                 .orElse(normalizeTag(rawGitBranch));
 
         JSONObject payload = new JSONObject();
+        payload.put("payload_version", buildData.getVersion());
         payload.put("level", PipelineStepData.StepType.PIPELINE.getBuildLevel());
         payload.put("url", buildData.getBuildUrl(""));
+
+        final long startTimeMillis = buildData.getStartTime(0L);
         payload.put("start", DatadogUtilities.toISO8601(new Date(startTimeMillis)));
-        payload.put("end", DatadogUtilities.toISO8601(new Date(endTimeMillis)));
+
+        Long endTime = buildData.getEndTime(null);
+        // there are valid cases where endTime is null (in-progress pipelines), and the backend allows for this
+        if (endTime != null) {
```

Comment:
Could we use the `isBuilding` boolean instead of `endTime` to check if the pipeline is running or not?

Reply:
Added `isBuilding` check